### ISSUE TITLE
Training stops after first epoch!!

### ISSUE THREAD

**Issue by iampartho** [galatolofederico/pytorch-balanced-batch#1]
I have using this method in my video classification as my dataset was hugely imbalance. But I have come across the issue of training being stopped after the first epoch. Since my validation set did not use balance batch therefore after the first epoch the code directly enters into validation or evaluation block. To deal with the problem after each epoch I reinitialize my train_dataloader variable and then it enters the next epoch for training. But re-initializing the dataloader takes about 1 hour in my system. This is why my training has become very slow. Can anyone help me with solving this issue other than re-initializing the train_dataloader?

**Comment by galatolofederico** [comment]
Hi thanks for using my code. I'll check out this issue in the weekend

**Comment by iampartho** [comment]
Thank you, waiting for hearing for you soon.

**Comment by galatolofederico** [comment]
Hey i'm back, I've checked out your issue. If I got it right you want to iterate multiple times over the same DataLoader using this sampler. This behaviour is possible since 5c5714668113c41c91713aea200f44e18eaa6ba1 (3 moths ago).
Look at `example.py`
```python
for epoch in range(0, epochs):
    for batch_x, batch_y in train_loader:
        print("epoch: %d labels: %s\ninputs: %s\n" % (epoch, batch_y, batch_x))
```
Here I loop over `train_loader` batch times without the need of re-initialization.

**Comment by iampartho** [comment]
Thank you for your reply. It did not quite solve the problem. It was not going into the training block of code from the 2nd epoch to the last epoch but after your solution it's still not going into training block of code in 2nd and 3rd epoch but it is going into training in 4th epoch then again it is not going into training in 5th and 6th epoch but again going into training on 7th epoch. Then skipping the next two epoch and again going into training on the third epoch and this is continuing. I do not know what am I suppose to do now.

**Comment by galatolofederico** [comment]
Probably there is something that I'm missing here, can you attach a Minimum Working Example? If you are concerned to publish your work online feel free to reach me in private at `federico.galatolo@ing.unipi.it`

**Comment by iampartho** [comment]
Yes I am planning for publishing my code. The thing is when I am working with small chunk of data it acts like the whole epoch is happening but when I do it with full data it just does for some batch then leave the epoch. I do not know how can I send you a minimum working example as the case totally depends upon total data?

**Comment by galatolofederico** [comment]
You can try to create a dummy version of your dataset, for example if you are working with 1000 images 800x600 rgb divided into 10 classes you can just use 
```
classes_prob = torch.tensor([0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1])
dataset_X = torch.randn(1000, 800, 600, 3)
dataset_Y = torch.distributions.categorical.Categorical(classes_prob.repeat(1000, 1)).sample()

dataset = torch.utils.data.TensorDataset(dataset_X, dataset_Y)
```

**Comment by iampartho** [comment]
If I send you the codes would you be able to check that out please ? I am using it for video classification, taking 40 frames per video.

**Comment by galatolofederico** [comment]
Sure! no problem
